lib/factory.rb

```diff
@@ -14,3 +16,140 @@
 # - to_a
 # - values_at
 # - ==, eql?
+# ----------------------------------------------------------------------------
+class Factory
+  def self.new(*fields, &block)
+    raise ArgumentError if fields.empty?
+
+    fields.each do |field|
```

Comment:
You don't need to loop through all fields. You can only check if the first argument is a constant or not.